wrappers/gtimelog_parser.py

```diff
@@ -18,65 +18,124 @@ def __init__(self, config):
+            for start, stop, duration, tags, entry in window.all_entries()
+            if self.is_dispatch_entry(entry)
+        ]) / len(dispatchable_entrie)
+
+    def get_issue_description(self, entry):
+        # remove comments
+        line = entry.split('|')[0]
+        try:
+            # remove category
+            if self.line_format == 'categorized':
+                # category is optional
+                if ':' in line:
+                    line = line.split(':', 1)[1].strip()
+            issue, description = [
+                x.strip() for x in line.split(self.delimiter, 1)
+                if x.strip()
+            ]
+        except ValueError:
+            print(
+                'Entry must be in the format `{}`. '
+                'Got '.format(self.line_format_str), entry
+            )
+            return
+
+        # no matter what we find as `issue`:
+        # if we have an alias override it takes precedence
+        if issue in self.aliases:
+            issue = self.aliases[issue]
+        return issue, description
+
     def get_entries(self, date_window):
         window = self.timelog.window_for(date_window.start, date_window.stop)
 
         worklogs = []
+        dispatchlogs = []
         attendances = []
+        dispatchable_entrie = self.get_dispatchable_entries(window)
+        duration_to_dispatch = self.get_duration_to_dispatch(window)
+
         for start, stop, duration, tags, entry in window.all_entries():
             if self.skip_entry(entry):
                 continue
             if attendances and attendances[-1][1] == start:
                 attendances[-1] = (attendances[-1][0], stop)
             else:
                 attendances += [(start, stop)]
-            # remove comments
-            line = entry.split('|')[0]
-            try:
-                # remove category
-                if self.line_format == 'categorized':
-                    # category is optional
-                    if ':' in line:
-                        line = line.split(':', 1)[1].strip()
-                issue, description = [
-                    x.strip() for x in line.split(self.delimiter, 1)
-                    if x.strip()
-                ]
-            except ValueError:
-                print(
-                    'Entry must be in the format `{}`. '
-                    'Got '.format(self.line_format_str), entry
-                )
-                continue
 
-            # no matter what we find as `issue`:
-            # if we have an alias override it takes precedence
-            if issue in self.aliases:
-                issue = self.aliases[issue]
-            worklogs.append(MultiLog(
+            issue, description = self.get_issue_description(entry)
+
+            dispatched_duration = int(duration.total_seconds())
+            if entry in dispatchable_entrie:
+                dispatched_duration = (
+                    int(duration.total_seconds()) + duration_to_dispatch
```

Comment:
does not seem to "equally" spread the amount across tasks... Either I don't understand the description or this piece of code :)

Reply:
Seems like a bug...

Reply:
Bug of course ^^

Reply:
Resolved

Reply:
Why not applying a percentage of the qty_to_dispatch instead of equally spreading it among dispatchable_entries?
see https://github.com/camptocamp/gtimelog2odoo/pull/13#issuecomment-1482539683
```python
dispatched_duration += (dispatched_duration / total_dispatchable_duration) * total_to_dispatch
```

Reply:
What worries me here is that, if you help someone for 15 minute, the customer will be invoiced the same amount of `to_dispatch_duration` than the customer on which you spent let's say 30 hour.
That doesn't seems very fair to me :)

Reply:
I see your though, but please don't forget a few things: 
 - We send timsheets each week (limited to 42h in CH for instance, if you work 30 hours in a row for a client, don't use the dispatch)
 - We are talking of durations between few minutes to some hours by week (personnaly, it's around 2h to 3h to dispatch with something like 20 to 40 effective entries)

So statistically, the problem you point on is mainly a theorical POV.
I dispatch at most 6 minutes in a whole week which is more than acceptable in terms of time saving if i need to do it myself.

cc @simahawk What you think ?

Reply:
```
>>> from math import ceil
>>> def round(val, precision=2):
...     multiplier = 10**precision
...     val *= multiplier
...     val = ceil(val)
...     return val / multiplier
... 
>>> round(1/3)
0.34
>>>  sum(round(1/3) for _ in range(3))
1.02
>>> round(1/3, precision=10)
0.3333333334
>>> sum(round(1/3, precision=10) for _ in range(3))
1.0000000002
```

Reply:
> * We send timsheets each week (limited to 42h in CH for instance, if you work 30 hours in a row for a client, don't use the dispatch)

This is not true. TS should be sent every day. In any case, this is what I and some others do.
AFAIG the time will be re-dispatched every time you push on a different number of entries.
Not a big deal but basically ALL the valid entries will be deleted and re-created _every_ time we push.
I don't think is a good idea. I'd like to think more about this :thinking:

Reply:
> PS: and btw, using percentages means to manage rounding which is a pain in the ass to properly do. I won't do it this way for this main reason. Make a proposal if you wish, we'll see this feature in an other PR to ease the merge of this one.

I wouldn't make it that complicated ATM. In any case this won't solve the issue I mentioned above.

Reply:
Seems like the relative simplicity i proposed start to become unsimplistic btw

Pushing timesheet each week solves everything (timesheet validation is done weekly, why wanting a different process between time push and time review is rational, i don't get it)

Reply:
Because: that's not what we were asked to do and it makes much sense.
However,  I don't want to open a discussion on this here. 
In any case, you cannot assume that everyone will push this only once a week. 
Hence, I'm not in favor of the current implementation.

Reply:
Ah BTW, one of the main reasons to not do it once a week, is that if you have errors and you have to fix them, you gonna lose much more time than when you spend 1 or max 5m per day to do it. That's statistical, I grant you ;)